### PR TITLE
Document S3 connector limitations

### PR DESCRIPTION
**Pull request by Sevenannn**
## 🗣 Description

Document S3 connector limitations. These limitations are Datafusion query physical planner limitation (i.e. Running these queries directly in Datafusion will fail) and there’s no mitigation for these queries.

### CODE REVIEW

File: crates/runtime/benches/queries/README.md [hunk 1]
```diff
@@ -406,5 +406,24 @@ select  cast(amc as FLOAT)/cast(pmc as FLOAT) am_pm_ratio
...
 | **Affected queries**     |                          |
 | ------------------------ | ------------------------ |
-| [q49.sql](tpcds/q49.sql) | [q75.sql](tpcds/q75.sql)|
-| [q90.sql](tpcds/q90.sql) |  |
+| [q49.sql](tpcds/q49.sql) | [q75.sql](tpcds/q75.sql) |
+| [q90.sql](tpcds/q90.sql) |                          |
+
+### Datafusion Physical Plan does not support logical expression `Exists`, `InSubquery` / not yet implemented for `GROUPING` aggregate function
+
+**Limitation**: Datafusion Physical Plan does not support `Exists`, `InSubquery`, `GROUPING` aggregate function, there will be errors when running the following queries in connector and in-memory arrow accelerator
```
Comment (phillipleblanc) [inline crates/runtime/benches/queries/README.md]: Which connector(s)?

Comment (Jeadie) [reply crates/runtime/benches/queries/README.md]: Is there any long-term or possible remedy?

Comment (Sevenannn) [reply crates/runtime/benches/queries/README.md]: Updated.

Comment (Sevenannn) [reply crates/runtime/benches/queries/README.md]: The long-term remedy would be having these aggregate functions implemented in Datafusion

**Review by sgrebnov — APPROVED** [review]

**Pull request merged**